## Decision Trees

In [12]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.tree import DecisionTreeRegressor
from sklearn.model_selection import train_test_split
from sklearn.cross_validation import cross_val_score
import pickle

In [13]:
train_data = pd.read_csv("training/goalie.csv")

In [14]:
rng = 42
X = train_data
y = pd.read_csv("ground_truth/goalie_y.csv") 

In [15]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.15, random_state = 10) 

In [16]:
regr_1 = DecisionTreeRegressor(max_depth=4)

In [17]:
model1 = regr_1.fit(X, y)

In [18]:
pickle.dump(regr_1,open("decision_tree_regressor.p","wb"))

In [19]:
scores1 = cross_val_score(model1,X_train,y_train,cv=5)

In [20]:
print (scores1)

[0.89650457 0.9214935  0.87512665 0.78945223 0.89834213]


In [21]:
y_1 = regr_1.predict(X)

In [22]:
print('Variance score: {}'.format(regr_1.score(X_test, y_test))) 

Variance score: 0.9212922271699598


## Results

For Decision trees regressor, the accuracies of ratings of players :

- **Goalkeeper** : 0.9212922271699598 , **Cross-val** = [0.89650457 0.9214935  0.87512665 0.78945223 0.90128941]

- **Forward** :0.8691558887107931, **Cross-val** = [0.79018471 0.76543679 0.82625728 0.80671281 0.76504952]

- **Defender** :0.8726119791337966, **Cross-val** = [0.77865767 0.73292849 0.79322629 0.7520389  0.79639222]

- **Midfielder** :0.8123021180974934, **Cross-val** = [0.84654581 0.79181928 0.80401843 0.79076226 0.82644766]

- **Overall** :0.7617652642782454,  **Cross-val** = [0.70637272 0.75799689 0.72172146 0.71281295 0.69522772]